src/test/scala/is/hail/variant/IntervalSuite.scala

```diff
@@ -239,16 +239,71 @@ class IntervalSuite extends SparkSuite {
     val gr37 = GenomeReference.GRCh37
     val gr38 = GenomeReference.GRCh38
 
-    val x = "GL000197.1:3739-GL000202.1:7538"
+    val x = "[GL000197.1:3739-GL000202.1:7538)"
     assert(Locus.parseInterval(x, gr37) ==
-      Interval(Locus("GL000197.1", 3739), Locus("GL000202.1", 7538)))
+      Interval(Locus("GL000197.1", 3739), Locus("GL000202.1", 7538), true, false))
 
-    val y = "HLA-DRB1*13:02:01:5-HLA-DRB1*14:05:01:100"
+    val y = "[HLA-DRB1*13:02:01:5-HLA-DRB1*14:05:01:100)"
     assert(Locus.parseInterval(y, gr38) ==
-      Interval(Locus("HLA-DRB1*13:02:01", 5), Locus("HLA-DRB1*14:05:01", 100)))
+      Interval(Locus("HLA-DRB1*13:02:01", 5), Locus("HLA-DRB1*14:05:01", 100), true, false))
 
-    val z = "HLA-DRB1*13:02:01:5-100"
+    val z = "[HLA-DRB1*13:02:01:5-100)"
     assert(Locus.parseInterval(z, gr38) ==
-      Interval(Locus("HLA-DRB1*13:02:01", 5), Locus("HLA-DRB1*13:02:01", 100)))
+      Interval(Locus("HLA-DRB1*13:02:01", 5), Locus("HLA-DRB1*13:02:01", 100), true, false))
+  }
+
+  @Test def testEndpoints() {
+    val ord = TInt32().ordering
+    val intContains = (i: Interval, v: Int) => i.contains(ord, v)
+
+    for (s1 <- Array(true, false); e1 <- Array(true, false)) {
+
+      val i0 = Interval(1, 1, includeStart=s1, includeEnd=e1)
+
+      assert(i0.isEmpty(ord) == (!i0.includeStart || !i0.includeEnd))
+
+      val i1 = Interval(1, 5, includeStart=s1, includeEnd=e1)
+
+      assert(i1.contains(ord, 3))
+      assert(i1.includeStart == i1.contains(ord, 1))
+      assert(i1.includeEnd == i1.contains(ord, 5))
+
```

Comment:
Can you add parser tests in here? Example parseInterval(i1.toString) == i1

Reply:
parser only parses locuses :( but I can try